controllers/common.go

```diff
@@ -60,12 +59,28 @@ func BuildCollector(params manifests.Params) ([]client.Object, error) {
 		}
 		resources = append(resources, objs...)
 	}
+	if params.OtelCol.Spec.TargetAllocator.Enabled {
+		taParams := targetallocator.Params{
+			Client:          params.Client,
+			Scheme:          params.Scheme,
+			Recorder:        params.Recorder,
+			Log:             params.Log,
+			Config:          params.Config,
```

Comment:
Given that these things are generic to any params, i wonder if it would be better to define a common type that always has these and the generic parameter is that type such that a TA params would just be 
```
type Params {
     manifestutils.Params
     Collector ...
     TargetAllocator ....
}
```